app/routes.py

```diff
@@ -1,2 +1,190 @@
+        completed_at=request_body["completed_at"]
+    )
+    db.session.add(new_task)
+    db.session.commit()
+    return jsonify({"task": new_task.task_dict()}), 201
+
+
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+    """ This endpoint takes in a GET request and possibly a sorting query and returns 
+    a sorted list of task instances or a non-sorted list of task instances """
+    sort_query = request.args.get("sort")
+    if sort_query == "desc":
+        tasks = Task.query.order_by(desc("title"))
+        task_list = [task.task_dict() for task in tasks]
+    elif sort_query == "asc":
+        tasks = Task.query.order_by(asc("title"))
+        task_list = [task.task_dict() for task in tasks]
+    else:
+        tasks = Task.query.all()
+        task_list = [task.task_dict() for task in tasks]
+    return jsonify(task_list), 200
+
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    """ This endpoint uses GET http method to find and return one specific instance of Task class """
+    one_task = Task.query.get(task_id)
+    if one_task is None:
+        return jsonify(one_task), 404
+    if one_task.goal_id:
+        return jsonify({"task": one_task.task_dict_with_goal()})
+    return jsonify({"task": one_task.task_dict()}), 200
+
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def put_one_task(task_id):
+    """ Edits the title and description information for a specific Task instance """
+    one_task = Task.query.get(task_id)
+    if one_task is None:
+        return jsonify(one_task), 404
+    request_body = request.get_json()
+    one_task.title = request_body["title"]
+    one_task.description = request_body["description"]
+    db.session.commit()
+    return jsonify({"task": one_task.task_dict()}), 200
+
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    """ Deletes a specific task instance based on the provided task id """
+    one_task = Task.query.get(task_id)
+    if one_task is None:
+        return jsonify(one_task), 404
+    db.session.delete(one_task)
+    db.session.commit()
+    response_body = f'Task {task_id} "{one_task.title}" successfully deleted'
+    return jsonify({"details": response_body}), 200
+
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_task(task_id):
+    """ Takes one specific task and adds datetime completion stamp to database for this task instance """
+    path = "https://slack.com/api/chat.postMessage"
+    one_task = Task.query.get(task_id)
+    if one_task is None:
+        return jsonify(one_task), 404
+    one_task.completed_at = datetime.now(tz=None)
+    requests.post(path, headers={"Authorization": os.getenv('BOT_TOKEN')},
+                  data={"channel": "slack-api-test-channel",
+                        "text": f"Someone just completed the task {one_task.title}"})
+    db.session.commit()
+    return jsonify({"task": one_task.task_dict()}), 200
```

Comment:
To follow the single responsibility principle, we can separate the logic for the slackbot into a helper function.